entities/storobj/storage_object.go

```diff
@@ -670,30 +680,33 @@ func DocIDAndTimeFromBinary(in []byte) (docID uint64, updateTime int64, err erro
-// 4          | uint32        | length of schema json
-// n          | []byte        | schema as json
-// 4          | uint32        | length of meta json
-// n          | []byte        | meta as json
-// 4          | uint32        | length of vectorweights json
-// n          | []byte        | vectorweights as json
-// 4          | uint32        | length of packed target vectors offsets (in bytes)
-// n          | []byte        | packed target vectors offsets map { name : offset_in_bytes }
-// 4          | uint32        | length of target vectors segment (in bytes)
-// n          | uint16+[]byte | target vectors segment: sequence of vec_length + vec (uint16 + []byte), (uint16 + []byte) ...
-// 4          | uint32        | length of multivectors as msgpack
-// n          | []byte        | multivectors as msgpack
+// No. of B      | Type                     | Content
+// --------------------------------------------------------------
+// 1             | uint8                    | MarshallerVersion = 1
+// 8             | uint64                   | index id, keep early so id-only lookups are maximum efficient
+// 1             | uint8                    | kind, 0=action, 1=thing - deprecated
+// 16            | uint128                  | uuid
+// 8             | int64                    | create time
+// 8             | int64                    | update time
+// 2             | uint16                   | VectorLength
+// n*4           | []float32                | vector of length n
+// 2             | uint16                   | length of class name
+// n             | []byte                   | className
+// 4             | uint32                   | length of schema json
+// n             | []byte                   | schema as json
+// 4             | uint32                   | length of meta json
+// n             | []byte                   | meta as json
+// 4             | uint32                   | length of vectorweights json
+// n             | []byte                   | vectorweights as json
+// 4             | uint32                   | length of packed target vectors offsets (in bytes)
+// n             | []byte                   | packed target vectors offsets map { name : offset_in_bytes }
+// 4             | uint32                   | length of target vectors segment (in bytes)
+// n             | uint16+[]byte            | target vectors segment: sequence of vec_length + vec (uint16 + []byte), (uint16 + []byte) ...
+// 4             | uint32                   | length of packed multivector offsets (in bytes)
+// n             | []byte                   | packed multivector offsets map { name : offset_in_bytes }
+// 4             | uint32                   | length of multivectors segment (in bytes)
+// 4 + (2 + n*4) | uint32 + (uint16+[]byte) | multivectors segment: num vecs + (vec length + vec floats), ...
+// TODO vec lengths immediately following num vecs so you can jump straight to specific vec?
```

Comment:
if this is a done TODO comment than can we remove it?

Reply:
It's just a reminder for me to talk with Andre and Andrzej about potential improvements to the file format (but I don't want to block this PR on them since we have some time before this is released)

Reply:
I'm happy to remove it or come back to it later

Reply:
let's keep it then